### PR TITLE
Fix consolelink

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
- consolelink should be applied to both self-managed and managed in downstream
- wrong code was introduced after previous rebase

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/dashboard/dashboard.go [hunk 1]
```diff
@@ -139,6 +139,27 @@ func (d *Dashboard) ReconcileComponent(owner metav1.Object, cli client.Client, s
 		if err != nil {
 			return fmt.Errorf("failed to set dashboard ISV from %s: %v", PathISVSM, err)
 		}
+		// ConsoleLink handling
+		consoleRoute := &routev1.Route{}
+		err = cli.Get(context.TODO(), client.ObjectKey{Name: "console", Namespace: "openshift-console"}, consoleRoute)
+		if err != nil && !apierrs.IsNotFound(err) {
+			return fmt.Errorf("Error getting console route URL : %v", err)
+		}
+		domainIndex := strings.Index(consoleRoute.Spec.Host, ".")
```
Comment (VaishnaviHire) [inline components/dashboard/dashboard.go:148]: Wouldn't this cause an error if console route  is not found?

Comment (zdtsw) [reply components/dashboard/dashboard.go:148]: yes, but we have `componentErrorList` and will continue to next component, right?
as for the dashboard itself, call to `DeployManifestsFromPath` are ahead of this part, should the configs should be applied

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**